ReactSkia/sdk/NotificationCenter.h

```diff
@@ -115,20 +120,22 @@ unsigned int NotificationCenter::on(std::string eventName, std::function<void (A
 
 template <typename... Args>
 void NotificationCenter::emit(std::string eventName, Args... args) {
-    std::list<std::shared_ptr<Listener<Args...>>> handlers;
-    
-    {
-        std::lock_guard<std::mutex> lock(mutex);
+    //Creating the dispatch Handler for dispach the event in the event base folly thread. 
+    auto dispatchHandler = [=](){   
+      std::list<std::shared_ptr<Listener<Args...>>> handlers;
+      {
+        std::lock_guard<std::mutex> lock(mutex);     
         auto range = listeners.equal_range(eventName);
         handlers.resize(std::distance(range.first, range.second));
-        std::transform(range.first, range.second, handlers.begin(), [] (std::pair<const std::string, std::shared_ptr<ListenerBase>> p) {
-            auto l = std::dynamic_pointer_cast<Listener<Args...>>(p.second);
-            
-            return l;
+        std::transform(range.first, range.second, handlers.begin(), 
+            [] (std::pair<const std::string, std::shared_ptr<ListenerBase>> p) {
+              auto l = std::dynamic_pointer_cast<Listener<Args...>>(p.second);  
+              return l;
         });
-    }
-
-    for (auto& h : handlers) {
+      }
+      for (auto& h : handlers) {
```

Comment:
should we add check for "h->cb" before calling, since in addListener we are adding the cb even if its null